README.md

```diff
@@ -95,20 +98,175 @@ await org.save()
+from pynocular.database_model import database_model, nested_model, UUID_STR
+
+from my_package import db_info
+
+@database_model("users", db_info)
+class User(BaseModel):
+
+    id: Optional[UUID_STR] = Field(primary_key=True, fetch_on_create=True)
+    username: str = Field(max_length=100)
+
+    created_at: Optional[datetime] = Field(fetch_on_create=True)
+    updated_at: Optional[datetime] = Field(fetch_on_update=True)
+
+@database_model("organizations", db_info)
+class Org(BaseModel):
+
+    id: Optional[UUID_STR] = Field(primary_key=True, fetch_on_create=True)
+    name: str = Field(max_length=45)
+    slug: str = Field(max_length=45)
+    # `organizations`.`tech_owner_id` is a foreign key to `users`.`id`
+    tech_owner: Optional[nested_model(User, reference_field="tech_owner_id")]
+    # `organizations`.`business_owner_id` is a foreign key to `users`.`id`
+    business_owner: nested_model(User, reference_field="business_owner_id")
+    tag: Optional[str] = Field(max_length=100)
+
+    created_at: Optional[datetime] = Field(fetch_on_create=True)
+    updated_at: Optional[datetime] = Field(fetch_on_update=True)
+
+
+tech_owner = await User.create("tech owner")
+business_owner = await User.create("business owner")
+
+
+# Creating org with only business owner set
+org = await Org.create(
+    name="org name",
+    slug="org-slug",
+    business_owner=business_owner
+)
+
+assert org.business_owner == business_owner
+
+# add tech owner
+org.tech_owner = tech_owner
+await org.save()
+
+# Fetch from the db and check ids
+org2 = Org.get(org.id)
+assert org2.tech_owner.id == tech_owner.id
+assert org2.business_owner.id == business_owner.id
+
+# swap user roles
+org2.tech_owner = business_owner
+org2.business_owner = tech_owner
+await org2.save()
+org3 = await Org.get(org2.id)
+assert org3.tech_owner.id == business_owner.id
+assert org3.business_owner.id == tech_owner.id
+
+
+# Serialize org
+org_dict = org3.to_dict()
+expected_org_dict = {
+    "id": org3.id,
+    "name": "org name",
+    "slug": "org-slug",
+    "business_owner_id": tech_owner.id,
+    "tech_owner_id": business_owner.id,
+    "tag": None,
+    "created_at": org3.created_at,
+    "updated_at": org3.updated_at
+}
+
+assert org_dict == expected_org_dict
+
+```
+
+When using `DatabaseModel.get(..)`, any foreign references will need to be resolved before any properties can be accessed that is not the primary id. If you try to access a property before calling `fetch()` on the nested model a `NestedDatabaseModelNotResolved` error will be thrown
+
+```python
+org_get = await Org.get(org3.id)
+org_get.tech_owner.id # Does not raise `NestedDatabaseModelNotResolved`
+org_get.tech_owner.username # Raises `NestedDatabaseModelNotResolved`
+
+org_get = await Org.get(org3.id)
+await org_get.tech_owner.fetch()
+org_get.tech_owner.username # Does not raise `NestedDatabaseModelNotResolved`
+```
+
+Alternatively, calling `DatabaseModel.get_with_refs` instead of `DatabaseModel.get` will
```

Comment:
To match "calling" conventions elsewhere
```suggestion
Alternatively, calling `DatabaseModel.get_with_refs()` instead of `DatabaseModel.get()` will
```